web-registry/src/lib/wallet/hooks/useOnAccountChange.tsx

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from 'react';
+
+import { Wallet } from '../wallet';
+import { WalletType } from '../walletsConfig/walletsConfig.types';
+import { ConnectWalletType } from './useConnectWallet';
+
+type Props = {
+  wallet: Wallet;
+  connectWallet: ConnectWalletType;
+};
+
+export const useOnAccountChange = ({ wallet, connectWallet }: Props): void => {
+  useEffect(() => {
+    if (typeof window === 'undefined') {
+      return;
+    }
+
+    const listener = (): void => {
+      if (wallet) {
+        connectWallet({ walletType: WalletType.Keplr });
+      }
+    };
+
+    window.addEventListener('keplr_keystorechange', listener);
+
+    return () => {
+      window.removeEventListener('keplr_keystorechange', listener);
```

Comment:
Listeners are meant to be removed when they are no longer needed because they use resources.

Reply:
So in this case, is the parent component the `WalletProvider`? In terms of the app, is there a particular action/s when that would get unmounted? thanks for taking the time to explain these things while i learn some react 🙏

Reply:
I agree here this should never happen since `WalletProdiver` is in the root app tree. That being said it's best to keep the cleaning function, it might be useful for example if the hook is used elsewhere later on. We might even have a lint for writing those cleaning functions.